src/java/org/apache/cassandra/gms/Gossiper.java

```diff
@@ -2027,6 +2022,9 @@ public List<String> reloadSeeds()
         seeds.addAll(tmp);
         // Remove the old entries
         seeds.retainAll(tmp);
+
+        seeds.remove(getBroadcastAddressAndPort());
+
```

Comment:
I'd propose
```
tmp.remove(getBroadcastAddressAndPort());
seeds.addAll(tmp);
seeds.retainAll(tmp);
```

This would prevent to get the own address in seeds even for a brief time.